docker/sui-node/Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM rust:1.60.0 AS chef
+WORKDIR sui
+RUN apt-get update && apt-get install -y cmake clang
+
+# Plan out the 3rd-party dependencies that need to be built.
+#
+# This is done by:
+#   1. Copy in Cargo.toml, Cargo.lock, and the workspace-hack crate
+#   2. Removing all workspace crates, other than the workpsace-hack
+#      crate, from the workspace Cargo.toml file.
+#   3. Update the lockfile in order to reflect the changes to the
+#      root Cargo.toml file.
+FROM chef AS planner
+COPY Cargo.toml Cargo.lock .
+COPY crates/workspace-hack crates/workspace-hack
+RUN sed -i '/crates\/workspace-hack/b; /crates/d' Cargo.toml \
+    && cargo metadata -q >/dev/null
```

Comment:
This step could potentially be done on the host machine, in an automate fashion, in the `build.sh` script which could save ~60s of time doing this computation (since updating the lockfile in this step requires downloading all third-party crates from crates.io + other git repos).